### PR TITLE
[Wait for #2109][FlatBuffer] Add nntrainer_schema.fbs @open sesame 03/13 15:00

### PR DESCRIPTION
**Pull request by DonghakPark**
### Add nntrainer_schema.fbs
- This schema for Flatbuffer export
- It contain only Fully-connected Layer Options
- It contain tensorflow lite's schema

### Modify meson_build
- For compiler nntrainer_schema_generated.h

**_This will be updated for more layers and operators_**

Signed-off-by: DongHak Park <donghak.park@samsung.com>

### CODE REVIEW

**Review by baek2sm — APPROVED** [review]
LGTM

File: nntrainer/compiler/nntrainer_schema.fbs [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+      beta2:float;
+      epsilon:float;
+}
+
+table Optimizer{
+      type:OptimizerType;
+      options:OptimizerOptions;
+}
+
+enum LossType:int32{
```
Comment (jihochu) [inline nntrainer/compiler/nntrainer_schema.fbs:302]:
```suggestion
enum LossType: int32{
```

File: nntrainer/compiler/nntrainer_schema.fbs [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+
+//Learning Rate
+enum LRSchedulerType: int32{
+     CONSTANT = 0,
+     EXPONENTIAL = 1,
+     STEP=2,
+}
+
+union LROptions{
```
Comment (jihochu) [inline nntrainer/compiler/nntrainer_schema.fbs:253]:
```suggestion
union LROptions {
```

File: nntrainer/compiler/nntrainer_schema.fbs [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+      SEGMENT_SUM = 125,
+      BATCH_MATMUL = 126,
+      PLACEHOLDER_FOR_GREATER_OP_CODES = 127,
+      CUMSUM = 128,
+      CALL_ONCE = 129,
+      BROADCAST_TO = 130,
+      RFFT2D = 131,
+      CONV_3D = 132,
+      IMAG=133,
+      REAL=134,
```
Comment (jihochu) [inline nntrainer/compiler/nntrainer_schema.fbs:145]:
```suggestion
      REAL = 134,
```

File: nntrainer/compiler/nntrainer_schema.fbs (hunk 1 above)
Comment (DonghakPark) [reply nntrainer/compiler/nntrainer_schema.fbs]: FIX - Thank you for Review

File: nntrainer/compiler/nntrainer_schema.fbs (hunk 2 above)
Comment (DonghakPark) [reply nntrainer/compiler/nntrainer_schema.fbs]: FIX - Thank you for Review

File: nntrainer/compiler/nntrainer_schema.fbs (hunk 3 above)
Comment (DonghakPark) [reply nntrainer/compiler/nntrainer_schema.fbs:145]: FIX - Thank you for Review

**Review by SeoHyungjun — APPROVED** [review]
LGTM!

It would be cleaner if the parts jihochu said were applied to other codes as well.

**Comment by DonghakPark** [comment]
> LGTM!
> 
> It would be neat to apply the fix Jihochu said to other code as well.

FIX --> Thank you for Review

File: nntrainer/compiler/meson.build [hunk 4]
```diff
@@ -37,6 +37,24 @@ if get_option('enable-tflite-interpreter')
     'tflite_interpreter.cpp',
     'tflite_opnode.cpp'
   ]
+
+if not flatc_prog.found()
```
Comment (jijoongmoon) [inline nntrainer/compiler/meson.build:41]: you might need the check the option related with tflite.

**Review by jihochu — APPROVED** [review]
LGTM

File: nntrainer/compiler/nntrainer_schema.fbs [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+     SIGMOID = 1,
+     SOFTMAX = 2,
+     RELU = 3,
+     LEAKY_RELU = 4,
+     NONE = 5,
+}
+
+//Tensor
+table Tensor {
+      type:TensorType;
```
Comment (jijoongmoon) [inline nntrainer/compiler/nntrainer_schema.fbs:199]: we might need format (NCHW or NHWC) later.

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**